docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,555 @@
+---
+title: Submitting a pull request to LoopBack 4
+lang: en
+keywords: contributing, LoopBack community, pull request, PR, loopback
+sidebar: contrib_sidebar
+permalink: /doc/en/lb4/submitting_a_pr.html
+---
+
+## How to submit a pull request (PR) for LoopBack 4
+
+Here are instructions on how to submit a pull request (PR) for LoopBack v4.
+
+If you are a github expert, please follow the
+[Expert instructions](#expert-instructions) below.
+
+If you are a github beginner, please follow the
+[Beginner Instructions](#beginner-instructions) below.
+
+### Expert Instructions
+
+In addition to your knowledge about github and creating a pull request, we have
+additional steps you need to follow when submitting a pull request for
+LoopBack 4.
+
+#### 1. Read the 'Contributing to LoopBack' documentation
+
+Be sure to read, understand, and abide by the instructions provided in
+[Contributing code](https://loopback.io/doc/en/contrib/code-contrib-lb4.html),
+[Contributing to docs](https://loopback.io/doc/en/contrib/doc-contrib.html), and
+[Contributing to LoopBack](https://loopback.io/doc/en/contrib/index.html).
+
+#### 2. Agree to the contributor license agreement (CLA)
+
+You must agree to the contributor license agreement (CLA) before the pull
+request can be `approved` and `merged` by the maintainers.
+
+Please read
+[Agreeing to the CLA](https://loopback.io/doc/en/contrib/code-contrib.html#agreeing-to-the-cla),
+click on this
+[link](https://cla.strongloop.com/agreements/strongloop/loopback-next), and sign
+it.
+
+#### 3. Ensure your changes are free of errors
+
+To ensure your `source code` files are free of formatting, syntax, and
+logical/execution errors, run :
+
+```
+npm run lint:fix && npm run test
+```
+
+To ensure your `documentation` files are free of formatting errors, run :
+
+```
+npm run lint:fix && npm run build:site
+```
+
+One important step of `build:site` is to convert documentation written in
+**Markdown** format into **HTML** format. To flush out any layout or
+substitution issues ( from using 'includes' for example), it is important to
+view these generated **HTML** files.
+
+To start a local web server that hosts the HTML documentation, run :
+
+```
+cd sandbox/loopback.io/
+npm start
+```
+
+In your browser, navigate to the specified server address, click on `Docs`,
+click on `LoopBack 4`, and verify that your documentation changes appear and are
+well-rendered.
+
+#### 4. Conventional Commits
+
+Our commit messages are formatted according to
+[Conventional Commits](https://conventionalcommits.org/).
+
```

Comment:
We should advise contributors to run `npm i` for `loopback-next`. It will create a `pre-commit` hook under `loopback-next/.git/pre-commit`, which enforces the conventional commit rules automatically.